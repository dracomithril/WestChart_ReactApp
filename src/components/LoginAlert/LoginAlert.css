.btn-login{
    padding: 0;
    border-radius: 4px;
    border-width: 0;
    margin-bottom: 5px;
    margin-top: 10px;
}
.btn-login a{
    margin-bottom: 0;
    margin-top: 0;
}