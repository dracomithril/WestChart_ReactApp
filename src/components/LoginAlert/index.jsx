/**
 * Created by Gryzli on 28.01.2017.
 */
import React from 'react';
import {Alert} from 'react-bootstrap';
import FacebookLogin from 'react-facebook-login';
import './LoginAlert.css';

export default class LoginAlert extends React.Component {
    render() {
        const prod_app_id = "1173483302721639";
        const test_app_id = "1173486879387948";
        return (<Alert bsClass="login-alert">
            <h4>Oh snap! You got an error!</h4>
            <p>{this.props.alertMessage}</p>
            <FacebookLogin
                appId={process.env.NODE_ENV === 'production' ? prod_app_id : test_app_id}
                language="pl_PL"
                autoLoad={true}
                scope="public_profile,email,user_managed_groups"
                callback={this.props.loginUser}
                fields="id,email,name,first_name,picture,groups{administrator}"
                cssClass="btn btn-social btn-facebook"
                icon={"fa fa-facebook"}
                version="v2.8"
            />
        </Alert>)
    }
}
LoginAlert.propTypes = {
    loginUser: React.PropTypes.func,
    alertMessage:React.PropTypes.string
};