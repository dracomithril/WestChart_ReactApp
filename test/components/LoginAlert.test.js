/**
 * Created by XKTR67 on 4/19/2017.
 */
import React from 'react';
import PropTypes from 'prop-types';
import LoginAlert from '../../src/front/components/LoginAlert/index';
import {shallow} from 'enzyme';
import {shallowToJson} from 'enzyme-to-json';
import configureMockStore from 'redux-mock-store';
const initial_state =require('../data/initial_state.json');
const mockStore = configureMockStore([]);
describe('<LoginAlert/>', () => {
    it('renders without crashing ChartPresenter', () => {
        const store = mockStore(initial_state);
        const wrapper = shallow(
            <LoginAlert />, {
                context: {store},
                childContextTypes: {store: PropTypes.object}
            }
        );
        expect(shallowToJson(wrapper)).toMatchSnapshot();

    });

});
