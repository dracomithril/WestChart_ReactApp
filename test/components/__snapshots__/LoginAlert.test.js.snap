// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginAlert/> renders without crashing ChartPresenter 1`] = `
<Jumbotron
  bsClass="login-info"
  componentClass="div"
>
  <h4>
    To start working witch us you need to login to facebook and spotify.
    <i
      className="fa fa-heart"
    />
  </h4>
  <p>
    Sorry you are not admin of this group.
  </p>
  <t
    appId="1173486879387948"
    autoLoad={true}
    callback={[Function]}
    cookie={false}
    cssClass="btn btn-social btn-facebook"
    disableMobileRedirect={false}
    fields="id,email,name,first_name,picture,groups{administrator}"
    icon="fa fa-facebook"
    isMobile={null}
    language="pl_PL"
    reAuthenticate={false}
    redirectUri="http://localhost/"
    scope="public_profile,email,user_managed_groups"
    size="metro"
    tag="button"
    textButton="Login with Facebook"
    typeButton="button"
    version="v2.8"
    xfbml={false}
  />
  <Button
    active={false}
    block={false}
    bsClass="btn"
    bsStyle="default"
    className="btn btn-social btn-spotify"
    disabled={false}
    onClick={[Function]}
  >
    <i
      className="fa fa-spotify"
    />
    Login to spotify
  </Button>
</Jumbotron>
`;
